### ISSUE TITLE
OpenRGB and/or E1.31 output support?

### ISSUE THREAD

**Issue by CalcProgrammer1** [cnlohr/colorchord#117]
This project looks pretty awesome!  I am the creator of an open source RGB lighting control app for PC hardware and peripherals called OpenRGB.  I was referred to your project by a user who wanted to be able to use colorchord's visualization on OpenRGB-supported hardware.  We have a fairly extensive list of supported hardware including keyboards, mice, motherboards, RAM, graphics cards, LED strip controllers, and more.

OpenRGB provides a network-based SDK for controlling lights, but I'm also working on an OpenRGB plugin for receiving E1.31 data (commonly used by other lighting software such as Vixen, XLights, and more).  Does colorchord have the option to output E1.31 already?  I couldn't find anything on this.  If you want to implement E1.31, I'd recommend the libe131 library.  It's worked very well for me.

OpenRGB: https://gitlab.com/CalcProgrammer1/OpenRGB

libe131: https://github.com/hhromic/libe131

**Comment by CaiB** [comment]
Currently, colorchord only supports raw UDP output. Packets containing nothing more than RGB data of each LED in series, optionally prepended by a specified number of padding bytes.

Similar to #112, making a new output mode wouldn't be particularly difficult, but you'll have to wait for cnlohr to see if he wants to implement it.

**Comment by CalcProgrammer1** [comment]
Raw RGB data shouldn't be too bad to read in.  I could write a plugin on my end to input this data, but at the same time E1.31 is very common and I think it would be a good addition here.

**Comment by CaiB** [comment]
@CalcProgrammer1 I have added experimental E1.31 output support to my ColorChord rewrite, [ColorChord.NET](https://github.com/CaiB/ColorChord.NET). I wanted to test it with OpenRGB, but I think I need a plugin, [this one?](https://gitlab.com/CalcProgrammer1/OpenRGBE131ReceiverPlugin) I can't find anywhere to download binaries for it. Where can I get them?

Thanks!

**Comment by CalcProgrammer1** [comment]
Yeah, you will need that plugin.  There is a GitLab CI set up that
automatically builds.  The current version should work with OpenRGB 0.6,
but will not work with latest master (need to update the submodule in the
plugin).

**Comment by CalcProgrammer1** [comment]
Actually, you should use the OpenRGBDevelopers repo, my personal fork is stale.

https://gitlab.com/OpenRGBDevelopers/OpenRGBE131ReceiverPlugin

The latest commit there should work with 0.6.

**Comment by CaiB** [comment]
Got it working! Thanks for your help.


https://user-images.githubusercontent.com/6701127/125508153-d1173318-e881-4e8a-967c-ac3f7be8b3e2.mp4
